src/lib/geoip/client.ts

```diff
@@ -126,40 +133,53 @@ export default class GeoipClient {
 			};
 		}
 
-		const maxmind = sources.find(s => s.provider === 'maxmind' && s.city === best.city);
-
-		if (maxmind?.asn && maxmind?.network) {
-			return {
-				asn: maxmind.asn,
-				network: maxmind.network,
-				normalizedNetwork: maxmind.normalizedNetwork,
-			};
+		for (const source of rankedSources) {
+			if (source.normalizedCity === best.normalizedCity && source?.asn && source?.network) {
+				return {
+					asn: source.asn,
+					network: source.network,
+					normalizedNetwork: source.normalizedNetwork,
+				};
+			}
 		}
 
 		return undefined;
 	}
 
-	private bestMatch (field: keyof LocationInfo, sources: LocationInfoWithProvider[]): LocationInfo {
+	private bestMatch (field: keyof LocationInfo, sources: LocationInfoWithProvider[]): [LocationInfo, LocationInfoWithProvider[]] {
+		const DESC_PRIORITY_OF_PROVIDERS: Provider[] = [ 'ip2location', 'ipmap', 'maxmind', 'ipinfo', 'fastly' ];
 		const filtered = sources.filter(s => s[field]);
-		// Group by the same field value
-		const grouped = Object.values(_.groupBy(filtered, field));
-		// Move items with the same values to the beginning
-		const ranked = grouped.sort((a, b) => b.length - a.length).flat();
+		// Initially sort sources so they are sorted inside groups
+		const sorted = filtered.sort((sourceA, sourceB) => {
+			const indexSourceA = DESC_PRIORITY_OF_PROVIDERS.indexOf(sourceA.provider);
+			const indexSourceB = DESC_PRIORITY_OF_PROVIDERS.indexOf(sourceB.provider);
+			return indexSourceA - indexSourceB;
+		});
```

Comment:
The priority is already established by the order in https://github.com/jsdelivr/globalping/pull/382/files#diff-026ba18a99076807dbf631c38a550192cb83dcbeb30cbb3521f9539bb8eedac9R56-R62 so I don't think this is necessary.